[DR-3361] Only log events to Bard when enabled

okotsopoulos opened the pull request:
https://broadworkbench.atlassian.net/browse/DR-3361

Possible contributor to degraded DrsHub performance following initial 8/15/23 load tests: we now log successful DRS resolutions in Bard.

To help verify the extent of Bard event logging’s impact on performance, we want to selectively disable it in DrsHub dev for additional scale tests.

Logging events to Bard is now controlled by new property `drshub.bardEventLoggingEnabled` which defaults to true, and can be overridden via environment variable `BARD_EVENT_LOGGING_ENABLED`.

okotsopoulos committed:
[DR-3361] Only log events to Bard when enabled

Controlled by new property drshub.bardEventLoggingEnabled which defaults to true, and can be overridden via environment variable BARD_EVENT_LOGGING_ENABLED

pshapiro4broad commented on service/src/main/java/bio/terra/drshub/config/DrsHubConfigInterface.java:
```diff
@@ -36,4 +36,8 @@ public interface DrsHubConfigInterface {
+
+  // Only log events to Bard when enabled: this is exposed to help debug performance regressions
+  // following the introduction of Bard logging.
+  Boolean bardEventLoggingEnabled();
```
This is following the existing convention, but I don't see why this interface uses boxed types. It looks like the generated code (`DrsHubConfig`) requires that a property is set before its getter is called, so it will always have a non-null value. Something to consider for a future PR.

okotsopoulos replied:
Yeah, good question. I wasn't sure why it was doing that either but opted to follow convention, since there's a good chance we'll ultimately remove this property support anyway.

okotsopoulos committed:
Expand TrackingInterceptorTest to ensure that disabling Bard emissions works

The test previously used a live DrsHubConfig: I refactored slightly to mock it so that we could manipulate it.
Noting that this unit test is more of a functional test that tests many components at once.
With more time I might be inclined to break up the testing into different unit test suites, but this does offer us fast, not-flaky coverage so I'll leave it as is for now.
